contracts/kernel/Kernel.sol

```diff
@@ -26,30 +28,44 @@ contract Kernel is IKernel, KernelStorage, Initializable, AppProxyFactory, ACLSy
         _setApp(APP_ADDR_NAMESPACE, ACL_APP_ID, acl);
 
         acl.initialize(_permissionsCreator);
+
+        recoveryVaultId = keccak256(APP_ADDR_NAMESPACE, apmNamehash("vault"));
     }
 
     /**
     * @dev Create a new instance of an app linked to this kernel and set its base
     *      implementation if it was not already set
     * @param _name Name of the app
     * @param _appBase Address of the app's base implementation
+    * @param _setDefault Whether the app proxy app is the default one.
+    *        Useful when the Kernel needs to know of an instance of a particular app,
+    *        like Vault for escape hatch mechanism.
     * @return AppProxy instance
     */
-    function newAppInstance(bytes32 _name, address _appBase) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
+    function newAppInstance(bytes32 _name, address _appBase, bool _setDefault) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
         _setAppIfNew(APP_BASES_NAMESPACE, _name, _appBase);
         appProxy = newAppProxy(this, _name);
+        // By calling setApp directly and not the internal functions, we make sure the params are checked
+        // and it will only succeed if sender has permissions to set something to the namespace.
+        if (_setDefault) setApp(APP_ADDR_NAMESPACE, _name, appProxy);
     }
 
     /**
     * @dev Create a new pinned instance of an app linked to this kernel and set
     *      its base implementation if it was not already set
     * @param _name Name of the app
     * @param _appBase Address of the app's base implementation
+    * @param _setDefault Whether the app proxy app is the default one.
+    *        Useful when the Kernel needs to know of an instance of a particular app,
+    *        like Vault for escape hatch mechanism.
     * @return AppProxy instance
     */
-    function newPinnedAppInstance(bytes32 _name, address _appBase) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
+    function newPinnedAppInstance(bytes32 _name, address _appBase, bool _setDefault) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
```

Comment:
@bingen I'm thinking it might be nice to have an overloaded version of `newAppInstance()` and `newPinnedAppInstance()` that just passes false to `setDefault` to avoid making this a breaking change.